### ISSUE TITLE
Some c++1z sneaking into boost.simd

### ISSUE THREAD

**Issue by alankelly** [agenium-scale/boost.simd#328]
In boost/simd/arch/common/scalar/function/log.hpp, line 103 and others, there are c++17 hexadecimal floating constants, 
This error was generated using g++ 6.2 with -pedantic-errors

boost/simd/arch/common/scalar/function/log.hpp:103:14: error: use of C++1z hexadecimal floating constant
         x *= 0x1p25f;

**Comment by ccharly** [comment]
I'm not having this issue while compiling a little example like the following: (without `-std=c++11`)
```c++
int main()
{
  double x = 0.1f;
  x *= 0x1p25f;
  return 0;
}
```

Enabling `-std=c++11` gives me the error... But only with `g++-6`, the others work fine..

**Comment by jfalcou** [comment]
I think it's an GNU extension that everybody implemented *then* got standardized in C++17. pedantic reports it as it's indeed an extension. It's trivial to replace with the proper literal I guess.

**Comment by jtlap** [comment]
sorry for the error in fact the code was inspired from C code in which the constant was used, and as it compiled I jumped over the pb.

**Comment by jfalcou** [comment]
@jtlap  can you fix it ?

**Comment by jtlap** [comment]
I think so

**Comment by jfalcou** [comment]
Fixed by  fafd34b
